src/Language/PureScript/Pretty/JS.hs

```diff
@@ -58,9 +57,9 @@ literals = mkPattern' match'
     , return $ emit "}"
     ]
     where
-    objectPropertyToString :: (Emit gen) => Text -> gen
-    objectPropertyToString s | identNeedsEscaping s = string s
-                             | otherwise = emit s
+    objectPropertyToString :: (Emit gen) => PSString -> gen
+    objectPropertyToString s | identNeedsEscaping $ T.pack $ codePoints s = string s
```

Comment:
@hdgarrood 

> I can't remember if I mentioned this already but exporting `codePoints` without using `Maybe` in the return type makes me uneasy because it's not always possible to consider a `PSString` as a sequence of code points (as you've already pointed out elsewhere).

That's correct, but this implementation represents lone surrogates in the PSString input as the reserved code point with the same index in the String output. So the function is total, though a bit opaque. As discussed with @andyarvanitis above, I'll add an explanatory comment. I don't think a `Maybe` makes the function any better.

How do you feel about adding another exposed function called `hasLoneSurrogates` or `isValidUTF16`? Then the guard for `objectPropertyToString` could be even more readable.